### PR TITLE
feat(ChatContainer): fit the chat to the on-screen keyboard on mobile

### PR DESCRIPTION
**Pull request by i-doshechnikow**
## What

On mobile the prompt input and the disclaimer stay above the on-screen keyboard.

iOS Safari keeps the layout viewport at full height while the keyboard is open, so `100vh`, `100dvh` and `height: 100%` all leave the bottom of the chat hidden behind it. `ChatContainer` now tracks `window.visualViewport` and clamps its own height to the visible area; the message list scrolls inside the remaining space and re-pins to the last message when its viewport shrinks. Android Chrome resizes the layout viewport itself, so nothing is clamped there.

## Changes

- `useKeyboardViewportFit`: new hook tracking `visualViewport` resize/scroll and returning `isKeyboardOpen` + `maxHeight`; the pure `resolveKeyboardViewportFit` is covered by unit tests.
- `ChatContainer`: new `adjustToKeyboard` prop (default `true`, opt-out for hosts that handle the keyboard themselves, e.g. via `interactive-widget=resizes-content`), `_keyboard-open` modifier and `max-height` in mobile mode.
- Welcome screen swaps its large top padding for `--g-aikit-chat-container-mobile-keyboard-empty-container-padding` while the keyboard is open.
- `useSmartScroll` / `useVirtualStickToBottom`: re-pin to the bottom when the scroll viewport is resized.
- README section on the on-screen keyboard + the new theming token.

## Testing

- `npm run typecheck`, unit tests and lint are green.
- Draft until the behaviour is confirmed on a real iOS Safari / Android Chrome device.

### CODE REVIEW

File: src/hooks/useKeyboardViewportFit.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+): KeyboardViewportFit {
+    const {viewportHeight, viewportOffsetTop, layoutHeight, containerTop} = metrics;
+
+    // Android Chrome resizes the layout viewport together with the visual one, so the inset stays
+    // at zero there and the container is left alone - the browser has already done the work.
+    if (layoutHeight - viewportHeight < minInset) {
+        return CLOSED;
+    }
+
+    return {
```
Comment (ananas7) [inline src/hooks/useKeyboardViewportFit.ts:59]: Pinch zoom is indistinguishable from an open keyboard here: increasing `visualViewport.scale` also reduces `visualViewport.height` and fires `resize`. On mobile, zooming far enough will therefore enable the keyboard modifier, clamp the container height, and change the welcome-screen padding even though no keyboard is open. Could we include `visualViewport.scale` (and, if needed, editable-element focus) in the detection and add a test for this case?

File: src/hooks/useKeyboardViewportFit.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    metrics: KeyboardViewportMetrics,
+    minInset: number = KEYBOARD_MIN_INSET,
+): KeyboardViewportFit {
+    const {viewportHeight, viewportOffsetTop, layoutHeight, containerTop} = metrics;
+
+    // Android Chrome resizes the layout viewport together with the visual one, so the inset stays
+    // at zero there and the container is left alone - the browser has already done the work.
+    if (layoutHeight - viewportHeight < minInset) {
+        return CLOSED;
+    }
```
Comment (ananas7) [inline src/hooks/useKeyboardViewportFit.ts:57]: This comment and the README do not match current Android Chrome behavior. Since Chrome 108, the default is `interactive-widget=resizes-visual`: only the Visual Viewport shrinks, while the Layout Viewport remains unchanged, so this implementation will clamp the container on Android as well. Both viewports resize only with `interactive-widget=resizes-content` (and in some WebViews). Could we update the comment and README to describe the actual behavior and opt-out scenario?

Comment (i-doshechnikow) [reply src/hooks/useKeyboardViewportFit.ts:57]: You are right, the comment and the README were describing pre-108 Chrome. Both are rewritten: the layout viewport stays full-height on Chrome 108+ and Firefox 132+ as well, so the clamp does apply on Android, and the zero-inset early return is what covers pages that opted into `interactive-widget=resizes-content`.

File: src/hooks/useKeyboardViewportFit.ts (hunk 1 above)
Comment (i-doshechnikow) [reply src/hooks/useKeyboardViewportFit.ts:59]: Good catch. Instead of bailing out at `scale > 1`, the inset is now measured in layout pixels: `layoutHeight - viewportHeight * scale`. A zoomed page gives a near-zero inset and is left alone, while the keyboard is still detected at any scale, which matters on iOS Safari where focusing an input with a font size below 16px auto-zooms the page. Added tests for both cases. I left editable-element focus out for now: the scaled measurement already removes the false positive, and an `activeElement` check brings its own edge cases (focus inside an iframe, custom editors). Happy to add it if you prefer the extra signal.

**Review by ananas7 — APPROVED** [review]
